### PR TITLE
Dog list cubit

### PR DESCRIPTION
**Pull request by k1mka**
ready

### CODE REVIEW

File: lib/presentation/get_it.dart [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import 'package:array_names/data/repositories/repository.dart';
+import 'package:array_names/data/repositories/repository_impl.dart';
+import 'package:array_names/data/services/network_service.dart';
+import 'package:get_it/get_it.dart';
+
+final getIt = GetIt.instance;
```
Comment (zddarova) [inline lib/presentation/get_it.dart]: get it is not presentation layer, move to diff folder

File: lib/service_locator.dart [hunk 2]
```diff
@@ -1,13 +1,16 @@
-import 'package:get_it/get_it.dart';
-
 import 'data/repositories/repository.dart';
 import 'data/repositories/repository_impl.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
 import 'data/services/network_service.dart';
 
-final getIt = GetIt.instance;
+final _networkServiceProvider = Provider<NetworkService>(
```
Comment (zddarova) [inline lib/service_locator.dart]: pls add todo to remove it in the future to not forget

File: lib/service_locator.dart [hunk 3]
```diff
@@ -1,13 +1,16 @@
...
+  (ref) => NetworkServiceImpl(),
+);
 
-void setupServiceLocator() {
-  NetworkService service = NetworkServiceImpl();
-  getIt.registerSingleton<NetworkService>(service);
-  getIt.registerSingleton<Repository>(RepositoryImpl(service));
-}
+final repositoryProvider = Provider<Repository>(
```
Comment (zddarova) [inline lib/service_locator.dart]: same here
pls add todo to remove it in the future to not forget

File: lib/presentation/screens/dog_list_screen/dog_list_screen.dart [hunk 4]
```diff
@@ -1,17 +1,18 @@
...
+import '../../../business_logic/dog_list_cubit.dart';
 
 class DogsListScreen extends StatelessWidget {
   const DogsListScreen({Key? key}) : super(key: key);
 
   @override
   Widget build(BuildContext context) {
-    return Provider<DogListProvider>(
-      create: (_) => DogListProvider(getIt<Repository>()),
+    return BlocProvider(
```
Comment (zddarova) [inline lib/presentation/screens/dog_list_screen/dog_list_screen.dart]: add generic type here to define what type of BlocProvider we create here

File: lib/presentation/screens/dog_list_screen/dog_list_screen.dart [hunk 5]
```diff
@@ -1,17 +1,18 @@
-import 'package:array_names/business_logic/dog_list_provider.dart';
 import 'package:array_names/data/repositories/repository.dart';
+import 'package:array_names/presentation/get_it.dart';
 import 'package:array_names/presentation/screens/dog_list_screen/dog_list_layout.dart';
-import 'package:array_names/service_locator.dart';
 import 'package:flutter/material.dart';
-import 'package:provider/provider.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+import '../../../business_logic/dog_list_cubit.dart';
```
Comment (zddarova) [inline lib/presentation/screens/dog_list_screen/dog_list_screen.dart]: remove relative imports

File: lib/presentation/screens/images_screen/images_screen_layout.dart [hunk 6]
```diff
@@ -1,40 +1,41 @@
-import 'package:array_names/business_logic/images_provider.dart';
+import 'package:array_names/data/models/breed.dart';
+import 'package:array_names/presentation/widgets/images_widget.dart';
 import 'package:flutter/material.dart';
-import 'package:provider/provider.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
 
-import '../../../data/models/breed.dart';
+import '../../../business_logic/images_provider.dart';
```
Comment (zddarova) [inline lib/presentation/screens/images_screen/images_screen_layout.dart]: relative import

File: lib/business_logic/dog_list_provider.dart [hunk 7]
```diff
@@ -1,13 +1,22 @@
...
-
-  DogListProvider(this._repo);
-
-  Future<List<Breed>> getDogs() async {
-    final dogList = await _repo.fetchDogs();
-    return dogList;
-  }
-}
+// import 'package:array_names/data/models/breed.dart';
+// import 'package:array_names/data/repositories/repository.dart';
```
Comment (zddarova) [inline lib/business_logic/dog_list_provider.dart]: add todo to remove file

File: lib/business_logic/dog_list_cubit.dart [hunk 8]
```diff
@@ -0,0 +1,10 @@
+import 'package:array_names/data/models/breed.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import '../data/repositories/repository.dart';
+
+class DogListCubit extends Cubit<List<Breed>> {
```
Comment (zddarova) [inline lib/business_logic/dog_list_cubit.dart]: let's discuss it closely tomorrow, come to me when you'll be free

Comment (zddarova) [reply lib/business_logic/dog_list_cubit.dart]: there is not the best work with states here

File: lib/business_logic/dog_list_cubit.dart [hunk 9]
```diff
@@ -0,0 +1,10 @@
...
+  Repository repo;
+  DogListCubit(this.repo) : super([]);
+
+  void cubitFetchDogs() async => emit(await repo.fetchDogs());
```
Comment (zddarova) [inline lib/business_logic/dog_list_cubit.dart]: 👍

**Review by zddarova — CHANGES_REQUESTED** [review]

**Comment by zddarova** [comment]
and we MUST resolve merge conflicts here tomorrow @k1mka

File: lib/presentation/get_it.dart (hunk 1 above)
Comment (k1mka) [reply lib/presentation/get_it.dart]: +

File: lib/service_locator.dart (hunk 2 above)
Comment (k1mka) [reply lib/service_locator.dart]: +

File: lib/service_locator.dart (hunk 3 above)
Comment (k1mka) [reply lib/service_locator.dart]: +

File: lib/business_logic/images_provider.dart [hunk 10]
```diff
@@ -1,21 +1,23 @@
-import 'package:array_names/data/models/breed.dart';
-import 'package:array_names/data/repositories/repository.dart';
-import 'package:array_names/service_locator.dart';
-import 'package:flutter_riverpod/flutter_riverpod.dart';
+//TODO: УДАЛИТЬ ФАЙЛ
```
Comment (AndreyKomarichev10) [inline lib/business_logic/images_provider.dart]: delete file

File: lib/service_locator.dart [hunk 11]
```diff
@@ -8,6 +8,8 @@ final _networkServiceProvider = Provider<NetworkService>(
   (ref) => NetworkServiceImpl(),
 );
 
+//TODO: УДАЛИТЬ ПОЗЖЕ
```
Comment (AndreyKomarichev10) [inline lib/service_locator.dart]: delete

File: pubspec.yaml [hunk 12]
```diff
@@ -32,6 +32,8 @@ dependencies:
     sdk: flutter
   http: ^0.13.4
   flutter_riverpod: ^1.0.4
```
Comment (AndreyKomarichev10) [inline pubspec.yaml:34]: delete riverpod dependency

**Review by AndreyKomarichev10 — APPROVED** [review]

**Review by zddarova — APPROVED** [review]

File: pubspec.yaml (hunk 12 above)
Comment (k1mka) [reply pubspec.yaml:34]: later

**Pull request merged**